simulator/bypass/data_bypass.cpp

```diff
@@ -13,59 +13,74 @@
 #include <infra/ports/timing.h>
 
 
-void DataBypass::update_register_info( const MIPSInstr& instr, RegNum num,
-                                       const Cycle& cycle,
-                                       DataBypass::RegisterStage new_dst_stage)
+
+void DataBypass::trace_new_register( const MIPSInstr& instr, RegNum num)
 {
     auto& entry = scoreboard.get_entry( num);
 
-    if ( new_dst_stage == 0_RSG)
-    {
-        entry.first_execute_stage_cycle = cycle;
-        entry.is_bypassible = false;
-        
-        if ( instr.get_dst_num() == REG_NUM_HI_LO)
-        {
-            is_HI_master_DIVMULT = true;
-        }
-        else if ( num == REG_NUM_HI)
-        {
-            is_HI_master_DIVMULT = false;
-        }
+    entry.current_stage = 0_RSG; // first execute stage
 
-        if ( instr.is_conditional_move())
-            entry.ready_stage = DataBypass::RegisterStage::in_RF();
-        else
-            entry.ready_stage = instr.is_load() ? 1_RSG  // MEMORY
-                                                : 0_RSG; // EXECUTE
+    if ( instr.is_conditional_move())
+        entry.ready_stage = RegisterStage::in_RF();
+    else
+        entry.ready_stage = instr.is_load() ? 1_RSG  // MEMORY 
+                                            : 0_RSG; // EXECUTE
+
+    if ( instr.get_dst_num() == REG_NUM_HI_LO)
+    {
+        is_HI_master_DIVMULT = true;
+    }
+    else if ( num == REG_NUM_HI)
+    {
+        is_HI_master_DIVMULT = false;
     }
-    else if ( entry.first_execute_stage_cycle + new_dst_stage.get_latency_from_first_execute_stage() != cycle)
-        return;
 
-    entry.current_stage = new_dst_stage;
+    entry.is_bypassible = ( entry.current_stage == entry.ready_stage) ? true : false;
```

Comment:
` ? true : false` is completely redundant (comparison result is a Boolean object itself)